app/dashboard/models.py

```diff
@@ -2203,40 +2215,58 @@ def to_dict(self, activities=True, leaderboards=True, network=None, tips=True):
         """
         params = {}
         network = network or self.get_network()
+        bounties = self.bounties
+        fulfilled_bounties = self.get_fulfilled_bounties(network=network)
+        funded_bounties = self.get_funded_bounties(network=network)
+        orgs_bounties = None
 
-        query_kwargs = {'network': network}
+        if self.is_org:
+            orgs_bounties = self.get_orgs_bounties(network=network)
 
-        sum_eth_funded = self.get_eth_sum(sum_type='funded', **query_kwargs)
-        sum_eth_collected = self.get_eth_sum(**query_kwargs)
-        works_with_funded = self.get_who_works_with(work_type='funded', **query_kwargs)
-        works_with_collected = self.get_who_works_with(work_type='collected', **query_kwargs)
-        funded_bounties = self.get_funded_bounties(network=network)
+        sum_eth_funded = self.get_eth_sum(sum_type='funded', bounties=funded_bounties)
+        sum_eth_collected = self.get_eth_sum(bounties=fulfilled_bounties)
+        works_with_funded = self.get_who_works_with(work_type='funded', bounties=funded_bounties)
+        works_with_collected = self.get_who_works_with(work_type='collected', bounties=fulfilled_bounties)
 
         # org only
-        works_with_org = []
         count_bounties_on_repo = 0
         sum_eth_on_repos = 0
-        if self.is_org:
-            works_with_org = self.get_who_works_with(work_type='org', **query_kwargs)
-            count_bounties_on_repo = self.get_orgs_bounties(network=network).count()
-            sum_eth_on_repos = self.get_eth_sum(sum_type='org', **query_kwargs)
+        works_with_org = []
+        if orgs_bounties:
+            count_bounties_on_repo = orgs_bounties.count()
+            sum_eth_on_repos = self.get_eth_sum(bounties=orgs_bounties)
+            works_with_org = self.get_who_works_with(work_type='org', bounties=orgs_bounties)
+
+        total_funded = funded_bounties.count()
+        total_fulfilled = fulfilled_bounties.count()
 
+        role = 'newbie'
+        if total_funded > total_fulfilled:
+            role = 'funder'
```

Comment:
Are we handling this in two methods now? Should we adjust the original method to handle this differently versus repeating the code or split it off?

Reply:
@mbeacom 
We could split it, if you want 👍